src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -310,9 +310,16 @@ SHAMapStoreImp::run()
             state_db_.setLastRotated(lastRotated);
         }
 
+        bool const readyToRotate =
+            validatedSeq >= lastRotated + deleteInterval_ &&
+            canDelete_ >= lastRotated - 1 && !health();
+
         // Make sure we don't delete ledgers currently being
         // imported into the ShardStore
-        bool const waitForImport = [this, lastRotated] {
+        bool const waitForImport = [this, readyToRotate, lastRotated] {
+            if (!readyToRotate)
+                return false;
+
```

Comment:
Ugh. It just occurred to me that this can be made simpler by taking advantage of `&&` short circuiting:
```
       bool const waitForImport = readyToRotate &&
          [this, lastRotated] {
            if (auto shardStore = app_.getShardStore())
[...]
```
This way, the lambda doesn't even get called if the system is not ready to rotate.

Reply:
That's cool. Done